aptos-move/framework/aptos-framework/sources/zkid.move

```diff
@@ -0,0 +1,123 @@
+module aptos_framework::zkid {
+    use std::option;
+    use std::option::Option;
+    use std::signer;
+    use aptos_framework::system_addresses;
+
+    #[resource_group(scope = global)]
+    struct Group {}
+
+    #[resource_group_member(group = aptos_framework::zkid::Group)]
+    /// The 288-byte Groth16 verification key (VK) for the zkID relation.
+    struct Groth16VerificationKey has key, store {
+        /// 32-byte serialization of `alpha * G`, where `G` is the generator of `G1`.
+        alpha_g1: vector<u8>,
+        /// 64-byte serialization of `alpha * H`, where `H` is the generator of `G2`.
+        beta_g2: vector<u8>,
+        /// 64-byte serialization of `gamma * H`, where `H` is the generator of `G2`.
+        gamma_g2: vector<u8>,
+        /// 64-byte serialization of `delta * H`, where `H` is the generator of `G2`.
+        delta_g2: vector<u8>,
+        /// 64-byte serialization of `\forall i \in {0, 1}, gamma^{-1} * (beta * a_i + alpha * b_i + c_i) * H`, where `H` is the generator of `G1`.
+        gamma_abc_g1: vector<vector<u8>>,
+    }
+
+    #[resource_group_member(group = aptos_framework::zkid::Group)]
+    struct Configuration has key, store {
+        // No transaction can have more than this many zkID signatures.
+        max_zkid_signatures_per_txn: u16,
+        // How far in the future from the JWT issued at time the EPK expiry can be set.
+        max_exp_horizon: u64,
+        // The training wheels PK, if training wheels are on
+        training_wheels_pubkey: Option<vector<u8>>,
+    }
+
+    // genesis.move needs to initialize the devnet VK
+    friend aptos_framework::genesis;
+
+    public(friend) fun initialize(fx: &signer, vk: Groth16VerificationKey, constants: Configuration) {
```

Comment:
with `friend` governance can't call it.

Reply:
do we have to initialize with `vk` and `constants`? we have dedicated setter functions for then, no?

Reply:
now it seems to me we don't even need an initialize function...

Reply:
We have setters for governance proposals. Mark as resolved if addressed.